Source/DafnyLanguageServer/Workspace/DocumentDatabase.cs

```diff
@@ -96,7 +95,10 @@ IRelocator relocator
 
       if (VerifyOnOpen) {
         Verify(documentEntry, cancellationSource.Token);
+      } else {
+        documentEntry.EndVerification();
```

Comment:
Why do we mark verification as complete in this case? / What is this for?

Reply:
This is for tests that want to know whether to expect any more updates.

Reply:
I wonder if they could use the telemetry notification that we added in https://github.com/dafny-lang/dafny/commit/5641c36fc08c347a081bad8b0260ca770dfc0059 ? (Or maybe it would be good to test that notification using the EndVerification event

Reply:
I'm removing that telemetry event because I don't think it fires at the right time in this PR. It relied on the Completed event from an observable, but the `DafnyDocument.VerificationUpdates` Observable in this PR is not completed even when the LSP server is idle, because the user can still manually trigger more verification tasks.

Reply:
That's unfortunate; will there be any way to detect completion of verification for a given file?  I'm thinking of programmatic uses of Dafny, where you want to gather errors and warnings for a given file using the LSP server.

Reply:
"completion of verification", you mean where everything that could be verified has been? You can call `VerifyAllTasksAsync` and then `await DocumentEntry.Idle` to do that.

Reply:
This is from C#, but I meant for an LSP client.  Suppose that I want to say "the document has been parsed, resolved, and verified; what can I depend on?  Only the custom verification status messages, right?

It would be nice to have a "standard" telemetry message.  And it should be possible to compute it correctly, since we have state update with our custom status event.

Reply:
> Only the custom verification status messages, right?

What's wrong with using these? The "standard" telemetry message would require custom logic to detect as well right?

Reply:
Correct.  But most LSP clients I've worked with make it marginally simpler to handle telemetry messages than custom ones, and we have changed the custom messages in the past.  Having a consistent telemetry message for that is nice.  Not a blocker.